### PR TITLE
DCOS-13451: Simplify the runtime configuration

### PR DESCRIPTION
**Pull request by wavesoft**
This PR simplifies the runtime configuration, by reordering some fields in the form and hiding runtime selection in the advanced section.

image

image

### CODE REVIEW

**Comment by nLight** [comment]
run integration tests

File: plugins/services/src/js/components/forms/GeneralServiceFormSection.js [hunk 1]
```diff
@@ -52,7 +53,7 @@ const containerRuntimes = {
     )
   },
   [MESOS]: {
-    label: <span>{labelMap[MESOS]} <em>(experimental)</em></span>,
+    label: <span>{labelMap[MESOS]} <span className="badge badge-rounded badge-success">Experimental</span></span>,
```
Comment (nLight) [inline plugins/services/src/js/components/forms/GeneralServiceFormSection.js]: ❤️  it

File: plugins/services/src/js/components/forms/GeneralServiceFormSection.js [hunk 2]
```diff
@@ -126,23 +124,43 @@ class GeneralServiceFormSection extends Component {
     }
 
     return (
-      <p>
-        {'Need to run a service with multiple containers? '}
-        <a className="clickable" onClick={this.handleOpenConvertToPodModal}>
-          Add another container
-        </a>.
-      </p>
+      <div className="pod pod-narrow pod-short">
```
Comment (nLight) [inline plugins/services/src/js/components/forms/GeneralServiceFormSection.js]: Don't like how `pod pod-narrow pod-short` add padding here. Is it by design?

screen shot 2017-01-27 at 12 24 34

Comment (wavesoft) [inline plugins/services/src/js/components/forms/GeneralServiceFormSection.js]: Not really, but it looks better when collapsed
image

It also helps differentiate it from the rest of the form when expanded

image

Comment (jfurrow) [inline plugins/services/src/js/components/forms/GeneralServiceFormSection.js]: I think it's fine to add margin to the top of this `Need to run a service...` copy, for now. However I think it should only be on the top, so your classes would look like this:
`pod pod-short flush-horizontal flush-bottom`.

In other parts of the form, we're relying on the top margin of heading tags to give separation to the different sections. In the future, I think we should create a `FormSection` component (or something similar) to provide consistent margins, but definitely not part of this PR!

File: plugins/services/src/js/components/forms/ContainerServiceFormAdvancedSection.js [hunk 3]
```diff
@@ -0,0 +1,257 @@
+import React, {Component} from 'react';
```
Comment (jfurrow) [inline plugins/services/src/js/components/forms/ContainerServiceFormAdvancedSection.js]: Good call on separating this to its own component! 👍

**Comment by jfurrow** [comment]
Nice changes! There are conflicts but otherwise I think it looks great!

File: plugins/services/src/js/components/forms/GeneralServiceFormSection.js (hunk 2 above)
Comment (leemunroe) [inline plugins/services/src/js/components/forms/GeneralServiceFormSection.js]: It seems odd that "Need to run a service..." isn't left aligned.

image

Comment (jfurrow) [inline plugins/services/src/js/components/forms/GeneralServiceFormSection.js]: @leemunroe That will be fixed with the classes I provided above (specifically the `flush-horizontal` class).

**Comment by rcorral** [comment]
Adding hold merge as we want to user test this before merging.

**Comment by wavesoft** [comment]
@rcorral let me know if user testing is OK with this.

**Comment by leemunroe** [comment]
@wavesoft @sascala Can we remove the `More information` link from the `Mesos` option? Seems odd to have a link for this when we don't have one beside the other options. Ideally there is something about the Mesos option in the [docs](https://docs.mesosphere.com/1.9/usage/containerizers/) that we link to.

**Comment by rcorral** [comment]
@wavesoft - @leemunroe will be running it by a few people in the company and report back.

**Comment by sascala** [comment]
@rcorral that doc _should_ say more about the Mesos containerizer. Once it does, then that page can be linked to from the `Container Runtime` tooltip. 

I added this to the Jira; will be done by 1.9 GA at the latest: https://mesosphere.atlassian.net/browse/DOCS-1664

**Comment by wavesoft** [comment]
This PR is now closed in favour of #1930

**Pull request closed without merging**